src/sentry/web/frontend/project_settings.py

```diff
@@ -187,6 +200,24 @@ def clean_slug(self):
                                         'using that slug' % other.name)
         return slug
 
+    def clean_token(self):
+        token = self.cleaned_data.get('token')
+        if not token:
+            return
+        token_re = r'^[a-zA-Z0-9-]{1,40}$'
```

Comment:
Hmm, I guess if the goal is with basic auth, it'll probably need to be longer than 40 chars, and also accept a wider set of characters since it's base64 encoded.

Lemme revisit this and get a better pattern here. Thanks for bringing it to my attention. :)